### PR TITLE
Hello React Rails

### PR DESCRIPTION
**Pull request by xtrahuman**
>Highlighted Below are tasks carried out on this project
- [x] Created a new Rails app called 'hello-rails-react'.
- [x] Initialized project with Git.
- [x] Set up the project with webpacker and the react-rails gems .
- [x] Make sure that the project has a Postgres database set up.
- [x] Created a table for storing messages and created 5 different greetings.
- [x] Created an API endpoint that selects a random greeting from my table
- [x] Created a static view that will be the root of my app.
- [x] Created the App component with react-router and render it in the static view.
- [x] Created the Greeting component that will display a greeting. Set it up as a route in App component.
- [x] Created a store, an action, and a reducer that will connect to the API endpoint to get the random greeting.
- [x] Display the random greeting in the Greeting component.

### CODE REVIEW

File: app/controllers/v1/greetings_controller.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class V1::GreetingsController < ApplicationController
+  def index
+    @greetings = Greeting.all
+    render json: { greeting: @greetings }.to_json
+  end
```
Comment (ahmetbozaci) [inline app/controllers/v1/greetings_controller.rb]: - [ ]  API endpoint should return a random greeting was created. You did it in react files. Please change it.
[Requirement](https://github.com/microverseinc/curriculum-rails/blob/main/connect-frontend-frameworks/hello_world_with_react.md#:~:text=Create%20an%20API%20endpoint%20that%20selects%20a%20random%20greeting%20from%20your%20table%20(you%20will%20need%20a%20controller%20with%20an%20action%20and%20Rails%20route).)

File: app/javascript/components/HelloWorld.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    .then((data) => dispatch(getMessageSuccessAction(data)))
+    .catch((error) => console.log(error));
+  }
+
+    return (
+      <div>
+         <div>
+           Greeting: <button onClick={() => performAction()}>Get Messages</button>
+         </div>
+          <p> {state[Math.floor(Math.random() * state.length)].greet}</p>
```
Comment (ahmetbozaci) [inline app/javascript/components/HelloWorld.js]: - [ ] You should get a random greeting from the API not in here. Please change it.

File: config/database.yml [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+  # https://guides.rubyonrails.org/configuring.html#database-pooling
+  pool: <%= ENV.fetch("RAILS_MAX_THREADS") { 5 } %>
+
+development:
+  <<: *default
+  database: hello_rails_react_development
+  host: localhost
+  pool: 5
+  username: hello_rails_react
+  password: hello_rails_react
```
Comment (ahmetbozaci) [inline config/database.yml]: ### OPTIONAL
- [ ] It is always good practice not to expose your database login information. Consider using the env file for all sensitive information about your application.

**Review by ahmetbozaci — CHANGES_REQUESTED** [review]
# STATUS: Changes Required  ♻️
hello
@xtrahuman  ✋🏼 
#### There are some issues that you still need to work on to go to the next project but you are almost there!

___ 

### 💯  Nice to have 💯
___


### ♻️ Request changes ♻️

- [ ] _Check the comments under the review._

___

### Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- Your commit messages should always start with capital letters and they should be an imperative tense. I see some of them are not like that. Instead of writing `Added style or Created the file` you should write `Add style or Create the file`
- ***I write it as optional. But this is a [general requirement](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md#:~:text=The%20commit%20message%20should%20be%20written%20in%20the%20imperative%20tense%20and%20capitalized) for every project. So you should pay attention to this in your next projects.***

**OPTIONAL**

- You can change them by using `git-rebase`. If you want to know how to change ı leave a few resources.
- [Video - How to change commit messages](https://www.loom.com/share/00771f369b7b47fc968c019ecf0d3ba6)
- [Article - How to create good commit messages](https://www.freecodecamp.org/news/level-you-up-to-awesome-commit-messages-a85558cb90e8/)
___

### Cheers and Happy coding! 💻 👏

almost

___

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form.](https://airtable.com/shrQAqnBwek5a0O0s)_


_If anything seems to be unclear to you, reach out and ask anything by commenting on this Pull Request and mentioning me as [ahmetbozaci](https://github.com/ahmetbozaci), I will receive a notification and will respond._
___
[GitHub](https://github.com/ahmetbozaci) | [LinkedIn](https://linkedin.com/in/ahmetbozaci) | [Twitter](https://twitter.com/ahmtbozaci)

_Reviewed by [Ahmet Bozacı ](https://github.com/ahmetbozaci)_

**Review by hiromataba — APPROVED** [review]
# Status: Approved :tada:  :champagne:  :confetti_ball: 
Hi  :wave:   :wave: @xtrahuman  

200w 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

### To Highlight :1st_place_medal: 
- Your Pull Request has a proper title and description. ✔️  
- No Linter errors :heavy_check_mark: 
- Your README file looks very professional :+1: 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**